### ISSUE TITLE
Is there a way to make the fullCalendar('option', {  to not force a reload or call a fetch of events

### ISSUE THREAD

**Issue by MiguelQueiroz** [fullcalendar/fullcalendar#4160]
I believe this issue does not require a sample demo.

**TL;DR**
Is there a way to make the fullCalendar('option', {  to not force a reload or call a fetch of events since i already have them? 

**Detailed explanation:**
I have a solution to calculate the min/max time of the agenda taking into account that some events that are outside of the defined min/max will progressively decrease or increase the min/max.

Considering that the week view has 7 days. If the events starts in day 0 - 1day (e.g: from 23:00 ) and ends at day 0 (2:00AM) the event will cross 2 days.
To be able to see this event which is outside of the pre-defined bussiness hours, the agenda must start at 00:00:00 (day zero). This is what i call a crossing beforeRange event. The same thing can happen if the event starts at day 7 at (22:00) and ends in the 7+1 day (e.g. 10AM) in this case this is a afterRange event, because it would force the agenda max time to be 24.  

For events that are beforeRange and afterRange  this will make max=24:00:00 and min=00:00:00 because it spans across all week.

Because the calendar min/max is set to 9AM to 9PM  and there are some event exceptions my eventSources async ajax request looks like this (simplified)

```
    app.data.calendarSettings.eventSources = [
function(start, end, timezone, callback) {
    app.api_get("/3/appt/get.json", {
        token: app.data.token,
        start: start.format(),
        end: end.format(),
        timezone: timezone
    }, function(out) {
        callback(afterFetchEvents(out));
    }, function() {
        callback(afterFetchEvents(app.data.allEvents)); //use the last event object cache if an error during fetch/offline
    });
```

this allows me to pre process the events that will go into the calendar, by using the callback function:

```
afterFetchEvents = function(events) {
    $(events).each(function(i, v) {
        var eventDateTimeStart = $.fullCalendar.moment.parseZone(v.start);
        var eventDateTimeEnd = $.fullCalendar.moment.parseZone(v.end);
        var eventCrossesView = eventDateTimeEnd >= calendarView.start && eventDateTimeStart <= calendarView.end;
        var eventBeforeRange = eventDateTimeStart < calendarView.start;
        var eventAfterRange = eventDateTimeEnd > calendarView.end;
        var eventOutBounds = eventDateTimeStart.date() != eventDateTimeEnd.date();
        if (eventCrossesView) {
            //If the event starts in current date, it can end in the next day after midnight, for those events we should expand the max time of the calendar.
            if (eventBeforeRange && eventAfterRange) {
                minTime = 0;
                maxTime = 24;
            } else if (eventBeforeRange) {
                minTime = 0;
            } else if (eventAfterRange) {
                maxTime = 24;
            } else {
                _minTime = Number(eventDateTimeStart.format("H.mm"));
                _maxTime = Number(eventDateTimeEnd.format("H.mm"));
                minTime = _minTime < minTime ? _minTime : minTime;
                maxTime = _maxTime > maxTime ? _maxTime : maxTime;
            }
        }

    });

    updateCalendarMinMaxTimes(minTime, maxTime);
    app.data.allEvents = events; //cache the current event object, if becomes offline
    return events;
}
```

```
updateCalendarMinMaxTimes = function(minTime, maxTime) {

    if (app.methods.calendar.is_24H) {
        minTime = 0;
        maxTime = 24;
    } else {
        //round time integer hours
        minTime = Math.floor(minTime);
        maxTime = Math.ceil(maxTime);
    }

    //normalize to the 00:00:00 format
    minTime = minTime.toString().split(".");
    maxTime = maxTime.toString().split(".");

    minTimeH = minTime[0] < 10 ? ("0" + minTime[0]) : minTime[0].toString();
    minTimeM = minTime[1] ? (minTime[1] < 10 ? (minTime[1] + "0") : minTime[1].toString()) : "00";

    maxTimeH = maxTime[0] < 10 ? ("0" + maxTime[0]) : maxTime[0].toString();
    maxTimeM = maxTime[1] ? (maxTime[1] < 10 ? (maxTime[1] + "0") : maxTime[1].toString()) : "00";

    minTime = minTimeH + ":" + minTimeM + ":00";
    maxTime = maxTimeH + ":" + maxTimeM + ":00";

    //optimize here do not call a refresh if the values did not change, also to avoid looping
    if (app.data.calendarSettings.minTime == minTime && app.data.calendarSettings.maxTime == maxTime) {
        return;
    }


    app.data.calendarSettings.minTime = minTime;
    app.data.calendarSettings.maxTime = maxTime;

    app.data.calendar.fullCalendar('option', {
        minTime: app.data.calendarSettings.minTime,
        maxTime: app.data.calendarSettings.maxTime
    });

}
```

Here,  the **updateCalendarMinMaxTimes**  sets the options of min/max, if any min is less than the current min, and increasing max if there is a max greater than current max, resulting in the best case scenario.

_To calculate if events are outside business hours, requiring to increase / decrease max/min values i have to have already events on my side (my API always returns full day range)._

**Problem** In case **afterFetchEvents** function found the need to modify min/max values, the **updateCalendarMinMaxTimes** function will call **fullCalendar('option',**  which will cause a re-render / call to ajax request again, calling twice my API, despite being useless.

I propose an argument of the fullCalendar('option', to disable re-fetch of events

**Comment by acerix** [comment]
I tested this and changing the `minTime` does cause the events to render, but doesn't actually re-fetch the events:

https://codepen.io/anon/pen/PezXMZ?editors=0010

However, changing to another day loads the events for that day, as expected.

So can you please [runnable, stripped-down demonstration](http://fullcalendar.io/wiki/Reduced-Test-Cases/) of the issue.  I can't tell from that code why the events are re-fetching.

**Comment by MiguelQueiroz** [comment]
Hello, thanks for the codepen and the delayed answer but i had to debug the fullcalendar step by step to understand what was that would make the fetch events method to be called, since on my side i had a different behaviour from your codepen.

My findings is that the difference between my code and yours can be traced to the fact that i have   **lazyFetching:false**. By doing this, changing the options min/max will always trigger a fetch of events.

I have this turned off because i always want fresh events and i assume that when someone changes from the monthly view to a week or day, i dont know how much time elapsed on the user side, so it can happen that some events might have changed. User always expect that by changing views will always fetch new events. 

The issue here is that i use the set option to quickly set the min and max ( very fast switching) to adjust the start of the calendar view after previously fetching events. Because the switching comes from a previous fetch calculation that further updates the min/max, makes no sense to re-fetch events, since within that time range, freshness of events is not important since i already had them :)

**Comment by MiguelQueiroz** [comment]
Yes basically instead of lazy fetching false it would be a time duration
example 2000 milliseconds, and that would mean the cache would be invalid
and as such a new fetch takes place. Otherwise use the cache.

**Comment by arshaw** [comment]
We never really got a reduced test case for this, so I can't confirm that it's fixed, but a very related issue [has been fixed](https://github.com/fullcalendar/fullcalendar/issues/3923#issuecomment-647781183) in v5 so I assume this one has as well.
